js/5etools-main.js

```diff
@@ -1331,6 +1332,261 @@ const betteR205etoolsMain = function () {
+							</div>
+						</div>
+					`).appendTo($("body"));
+					const $remain = $dialog.find(`[name="remain"]`);
+					const $cbSkill = $dialog.find(`input[type="checkbox"]`);
+
+					$cbSkill.on("change", function () {
+						const $e = $(this);
+						let selectedCount = getSelected().length;
+						if (selectedCount > count) {
+							$e.prop("checked", false);
+							selectedCount--;
+						}
+						$remain.text(`Remaining: ${count - selectedCount}`);
+					});
+
+					function getSelected () {
+						return $cbSkill.map((i, e) => ({skill: $(e).data("prof"), selected: $(e).prop("checked")})).get()
+							.filter(it => it.selected).map(it => it.skill);
+					}
+
+					$dialog.dialog({
+						dialogClass: "no-close",
+						buttons: [
+							{
+								text: "Cancel",
+								click: function () {
+									$(this).dialog("close");
+									$dialog.remove();
+									reject(`User cancelled the prompt`);
+								}
+							},
+							{
+								text: "OK",
+								click: function () {
+									const selected = getSelected();
+									if (selected.length === count) {
+										$(this).dialog("close");
+										$dialog.remove();
+										resolve(selected);
+									} else {
+										alert(`Please select ${count} language${count === 1 ? "" : "s"}`);
+									}
+								}
+							}
+						]
+					})
+				});
+			}
+
+			async function chooseProfsGroup (options, profType) {
+				// For when there are two separate ways to choose languages
+				return new Promise((resolve, reject) => {
+					const $dialog = $(`
+						<div title="Choose ${profType}">
+							<div>
+								${options.map((it, i) => `<label class="split"><input name="prof-group" data-ix="${i}" type="radio" ${i === 0 ? `checked` : ""}> <span>${
+									// Format it nicely
+									Object.entries(it).map(a => a[0]).map(a => a === "anyStandard" ? "any" : a).map(a => a.toTitleCase()).join(", ")
+								}</span></label>`).join("")}
+							</div>
+						</div>
+					`).appendTo($("body"));
+					const $rdOpt = $dialog.find(`input[type="radio"]`);
+
+					$dialog.dialog({
+						dialogClass: "no-close",
+						buttons: [
+							{
+								text: "Cancel",
+								click: function () {
+									$(this).dialog("close");
+									$dialog.remove();
+									reject(`User cancelled the prompt`);
+								}
+							},
+							{
+								text: "OK",
+								click: function () {
+									const selected = $rdOpt.filter((i, e) => $(e).prop("checked"))
+										.map((i, e) => $(e).data("ix")).get()[0];
+									$(this).dialog("close");
+									$dialog.remove();
+									resolve(selected);
+								}
+							}
+						]
+					})
+				});
+			}
+
+			async function handleProfs(profs, profType) {
+				// Handle the language options, let user choose if needed
+				// Handles most edge cases I think
+				ret = []
+				const profEntries = Object.entries(profs);
+				for (const entry of profEntries) {
```

Comment:
could be an `Object.entries(profEntries).forEach(([key, value]) => { ... })`

Reply:
j/k I see there's an `await` in there : )

Reply:
...in which case it can be `for (const [key, value] of Object.entries(profs)) {`